subsys/bluetooth/shell/iso.c

```diff
@@ -33,7 +49,41 @@ static void iso_recv(struct bt_iso_chan *chan, const struct bt_iso_recv_info *in
 
 static void iso_connected(struct bt_iso_chan *chan)
 {
+	struct bt_iso_info iso_info;
+	int err;
+
 	shell_print(ctx_shell, "ISO Channel %p connected", chan);
+
+
+	err = bt_iso_chan_get_info(chan, &iso_info);
+	if (err != 0) {
+		printk("Failed to get ISO info: %d", err);
+		return;
+	}
+
+	if (iso_info.type == BT_ISO_CHAN_TYPE_CONNECTED) {
+		static bool cis_timer_initialized;
+
+		cis_sn = 0U;
+
+		if (!cis_timer_initialized) {
+			k_work_init_delayable(&cis_sn_work, sn_timeout);
+			cis_timer_initialized = true;
+		}
+
+		k_work_schedule(&cis_sn_work, K_USEC(cis_sdu_interval_us));
```

Comment:
The minimum SDU interval is 255us but the tick timer is typically between 1ms and 10ms - hence you will not have a stable timer firing every 255us - at least not when the system is busy. I think the only stable way to increment this counter is to take a timestamp here and calculate the increment every time send() is called.

Reply:
That's a good idea, basically calculating the (minimum) SN increment instead of continuously incrementing it by 1. Will update.

Reply:
Hmm, implemented it and it seems to work correctly. However, when we are still using 255us as the increment factor, then the values are incrementing quite fast (in fact the SN wraps every ~16 second), which makes it damn near impossible to use this with the SN verification that I implemented... 

Another solution would be to, for the ISO peripheral, to have a command that tells the shell what the SDU interval is and use that, and avoid the 255us increment altogether.